Assets/UPM/Console/CommandDiscovery.cs

```diff
@@ -55,15 +68,30 @@ public void Invoke(string methodName)
 
             Execute(commandMap[splitString[0]].Program, argumentList);
         }
+        else
+        {
+            Debug.Log("Cant find Command: " + splitString[0]);
+        }
     }
 
     private IEnumerable<string> GenerateArgumentList(IEnumerable<string> arguments)
     {
         return arguments;
     }
 
-    private void Execute(MethodInfo program, IEnumerable<string> arguments)
+    private void Execute(MethodInfo program, IEnumerable<string> arguments, Type activator = null)
     {
-        program.Invoke(null, arguments.ToArray());
+        if (program.IsStatic)
+        {
+            program.Invoke(null, arguments.ToArray());
+        }
+        else
+        {
+            //TODO fix non Static, because then we need to execute with the class that has this Methode/Program
+            object instance = Activator.CreateInstance(activator); // activator = Type = the class/Type that has the Methode ?
```

Comment:
this is the thing that I didnt fix last time I made a Console/Terminal back in Unity 5.3

the activator should be the Type of the class that has the methode (not sure)